test/asd-test/test-asd.js

```diff
@@ -0,0 +1,122 @@
+// should test the following:
+// 1. It should grab the test using 'grabTest()' from the selected project
+// 2. It should run the test using 'runTest()' from the selected project
+// 3. might have to create own pass /fail object structure for asd
+
+/* global describe it expect before beforeEach afterEach */
+require('mocha');
+require('should');
+const fs = require('fs');
+const rimraf = require('rimraf');
+const expect = require('chai').expect;
+const proxyquire = require('proxyquire');
+const changeCase = require('change-case');
+
+const fakeHelpers = require('../helpers/fakeHelpers');
+
+const {
+  dummySessionAsd,
+} = require('../helpers/dummyData');
+
+const projects = proxyquire('../../controller/projects', {
+  './env': {
+    home: fakeHelpers.home
+  }
+});
+
+const test = proxyquire('../../controller/test', {
+  './helpers': fakeHelpers,
+  './github': fakeHelpers,
+  './env': {
+    home: fakeHelpers.home
+  }
+});
+
+const projectsDirectory = './test/files/environment/projects';
+
+describe('test', function () {
+  before(function (done) {
+    rimraf(projectsDirectory, () => done());
```

Comment:
You probably want `rimraf` to invoke the `done` callback directly. `rimraf`, like all node-style async functions, calls the given callback with an error if there is one, or nothing. That is exactly what `done` expects, an error as an argument or nothing if everything is all good. So this is the easiest way to tell the mocha that there was an error invoking `rimraf.
```suggestion
    rimraf(projectsDirectory, done);
```

Reply:
@harveysanders  thanks for the explanation